packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -69,8 +68,12 @@ export class BlobManager {
         });
     }
 
+    private hasBlob(id: string): boolean {
+        return this.blobIds.has(id) || this.detachedBlobIds.has(id);
+    }
+
     public async getBlob(blobId: string): Promise<IFluidHandle<ArrayBufferLike>> {
-        assert(this.blobIds.has(blobId) || this.pendingBlobIds.has(blobId), 0x11f /* "requesting unknown blobs" */);
+        assert(this.hasBlob(blobId), 0x11f /* "requesting unknown blobs" */);
```

Comment:
This is an example of where we should replace this assert with an if and throw a more user-friendly error. I'm adding a new container error type "UsageError" for this purpose, but I haven't merged it. You could do new GenericError or leave it as-is for now.

Reply:
I would argue that an assert is appropriate here. For context, this is not really a public-facing API. `getBlob()` is used by `ContainerRuntime` to route blob handles. This assert being tripped would most likely indicate data loss.